src/quantum_gates/_utility/circ_optimizator.py

```diff
@@ -0,0 +1,491 @@
+"""
+This file implement a class that represents an optimization algorithm for the output of the circuit class. 
```

Comment:
@paolodr98 : Question: How did we come up with the optimizations? Are they just necessary to speed up the execution or do they come from a paper or our own ideas? 

At first, I thought the optimization level might be a Qiskit concept, so you might want to give some background information.

Reply:
It's an idea we came up with to speed up the execution of the new BinaryBackend; it takes advantage of the gate patterns in the transpiled circuit. It is not releated with the Qiskit optimization, (I imagined that the name would lead to misunderstandings, but it was the most fitting one at the time I wrote the code). We will write a more detailed documentation